### PR TITLE
feat: enhance logging

### PR DESCRIPTION
**Pull request by gallexis**
We can now run the program and specify log level via arg or env : 
- ```ENV LOG=debug go run . start```
- ```go run . -log=debug start```

3 levels available : trace, debug and info
- **trace** & **debug** for dev purpose with slow but clean logs
- **info** for prod with fast json encoded logs

Also, this PR will try to replace every fmt.print based logs with the zerolog lib, and will log all the silenced errors, at the cost that during the tests for exemple, too many lines might be displayed.

### CODE REVIEW

**Comment by tchardin** [comment]
@gallexis btw any zerolog should be used strictly for logging. So CLI command feedback/messages should stay with `fmt.Printf` yeah?

**Comment by gallexis** [comment]
> @gallexis btw any zerolog should be used strictly for logging. So CLI command feedback/messages should stay with `fmt.Printf` yeah?

Yeah it sounds good like that

**Comment by tchardin** [comment]
Is there a way to log the line where the log call is situated? Hard to know where it's coming from otherwise

**Comment by gallexis** [comment]
> Is there a way to log the line where the log call is situated? Hard to know where it's coming from otherwise

I added it thanks to a hook, but only for debug/trace level because it slows down the logging

**Comment by tchardin** [comment]
@gallexis any idea what is going on with the commit history here?

**Comment by gallexis** [comment]
Yeah idk what happened, I pulled the main branch then rebase with this branch

**Comment by gallexis** [comment]
I'll look monday how to get rid of these commit

File: filecoin/storage/adapter.go [hunk 1]
```diff
@@ -31,10 +32,10 @@ type Adapter struct {
 
 // GetChainHead returns a tipset token for the current chain head
 func (a *Adapter) GetChainHead(ctx context.Context) (shared.TipSetToken, abi.ChainEpoch, error) {
-	fmt.Println("GetChainHead")
+	log.Info().Msg("GetChainHead")
```
Comment (tchardin) [inline filecoin/storage/adapter.go:35]: These I will probably remove when storage is more stable. This was mostly to help debug stuff.

File: node/server.go [hunk 2]
```diff
@@ -164,7 +164,7 @@ func (s *server) getHandler(w http.ResponseWriter, r *http.Request) {
 	// try to retrieve the blocks
 	err = s.node.get(r.Context(), root, &GetArgs{Key: key, Strategy: "SelectFirst"})
 	if err != nil {
-		fmt.Printf("ERR %s\n", err)
+		log.Error().Err(err).Msg("error")
```
Comment (tchardin) [inline node/server.go:167]:
```suggestion
		log.Error().Err(err).Msg("failed to get blocks")
```

**Review by tchardin — APPROVED** [review]
This looks good. Curious why you upgraded ffcli to v3?

**Comment by gallexis** [comment]
Actually I just tried the v3 to see if there was any breaking change. Seems there is not so why not using the v3?

**Comment by tchardin** [comment]
Word, just curious

File: cmd/pop/cli/cli.go [hunk 3]
```diff
@@ -7,25 +7,81 @@ import (
...
+	if _, file, line, ok := runtime.Caller(3); ok {
+		e.Str("file", path.Base(file)).Int("line", line)
+	}
+}
+
 // Run runs the CLI. The args do not include the binary name.
 func Run(args []string) error {
-	if len(args) == 1 && (args[0] == "-V" || args[0] == "--version" || args[0] == "version") {
+	rootfs := flag.NewFlagSet("pop", flag.ExitOnError)
+	version := rootfs.Bool("version", false, "Display pop version")
```
[the remark this answers is not in the document]
Comment (gallexis) [reply cmd/pop/cli/cli.go:42]: Yeah for the Version I just reverted to the old way

File: exchange/replication.go [hunk 4]
```diff
@@ -139,9 +140,22 @@ func NewReplication(h host.Host, idx *Index, dt datatransfer.Manager, rtv Routed
 	}
 	r.hs = NewHeyService(h, pm, r)
 	h.SetStreamHandler(PopRequestProtocolID, r.handleRequest)
-	r.dt.RegisterVoucherType(&Request{}, r)
-	r.dt.RegisterTransportConfigurer(&Request{}, TransportConfigurer(r.idx, r, h.ID()))
-	r.emitter, _ = h.EventBus().Emitter(new(IndexEvt))
+
+	err := r.dt.RegisterVoucherType(&Request{}, r)
```
[the remark this answers is not in the document]
Comment (gallexis) [reply exchange/replication.go:144]: Yes I think we should log.Fatal(), which will quit ( os.Exit(1) ) the program right after logging the error. We might instead do a log.Panic() that will also display the stack trace in the log before quitting, what do you think?

[the remark this answers is not in the document]
Comment (tchardin) [reply exchange/replication.go:144]: I think just returning the error is more idiomatic. We're not at the top level and this package could be consumed by other implementations

**Comment by tchardin** [comment]
I'm getting more and more confused about his commit history hahaha

**Comment by gallexis** [comment]
ahah don't worry I think I fixed it by created a new branch a rebasing it without only my commits

**Pull request closed without merging**